util/src/dbutil.cpp

```diff
@@ -281,39 +290,41 @@ std::shared_ptr<odb::database> createDatabase(const std::string& connStr_)
 
   std::vector<std::string> odbOpts = createOdbOptions(options);
   char** cStyleOptions = createCStyleOptions(odbOpts);
+  int optionsSize = odbOpts.size();
 
   std::shared_ptr<odb::database> db;
 
-#ifdef DATABASE_MYSQL
-  if (database == "mysql")
-    db.reset(new odb::mysql::database(odbOpts.size(), cStyleOptions),
-      [](odb::database*){});
-#endif
-
 #ifdef DATABASE_SQLITE
   if (database == "sqlite")
   {
-    int optionsSize = odbOpts.size();
-    auto sqliteDB = new odb::sqlite::database(
-      optionsSize,
-      cStyleOptions,
-      false,
-      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE,
-      true,
-      "",
-      std::make_unique<odb::sqlite::single_connection_factory>());
-    db.reset(sqliteDB, [](odb::database*){});
-
-    auto sqlitePtr = sqliteDB->connection()->handle();
-    sqlite3_create_function_v2(
-      sqlitePtr,
-      "regexp", 2, // regexp function with one argument
-      SQLITE_UTF8,
-      nullptr,
-      &sqliteRegexImpl,
-      nullptr,
-      nullptr,
-      nullptr);
+    try
+    {
+      auto sqliteDB = new odb::sqlite::database(
+        optionsSize,
+        cStyleOptions,
+        false,
+        SQLITE_OPEN_READWRITE | (create_ ? SQLITE_OPEN_CREATE : 0),
+        true,
+        "",
+        std::make_unique<odb::sqlite::single_connection_factory>());
+      db.reset(sqliteDB, [](odb::database*){});
+
+      auto sqlitePtr = sqliteDB->connection()->handle();
+      sqlite3_create_function_v2(
+        sqlitePtr,
+        "regexp", 2, // regexp function with one argument
+        SQLITE_UTF8,
+        nullptr,
+        &sqliteRegexImpl,
+        nullptr,
+        nullptr,
+        nullptr);
+    }
+    catch(odb::database_exception& e)
```

Comment:
Space

Reply:
Fixed in 64cb407581662e78675baf916a95dba2b765b3e4.